File: Greedy/Medium/LET_no134_Gas Station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        n = len(gas)
        diff = [gas[i] - cost[i] for i in range(n)]
        if sum(diff) < 0:
            return -1
        
        cur, start = 0, 0
        for i in range(n):
            cur += diff[i]
            if cur < 0:
                cur = 0
                start = i + 1
        return start

File: Greedy/Medium/LET_no134_Gas Station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int n = gas.size();
        for (int i = 0; i < n; i++)
            gas[i] -= cost[i];

        if (accumulate(gas.begin(), gas.end(), 0) < 0)
            return -1;

        int start = 0, cur = 0;
        for (int i = 0; i < n; i++) {
            cur += gas[i];
            if (cur < 0) {
                cur = 0;
                start = i + 1;
            }
        }

        return start;
    }
};